core/src/main/java/org/apache/carbondata/core/datastore/compression/ZstdCompressor.java

```diff
@@ -0,0 +1,200 @@
+ */
+
+package org.apache.carbondata.core.datastore.compression;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+
+import org.apache.carbondata.common.logging.LogService;
+import org.apache.carbondata.common.logging.LogServiceFactory;
+
+import com.github.luben.zstd.Zstd;
+
+public class ZstdCompressor implements Compressor, Serializable {
+  private static final long serialVersionUID = 8181578747306832771L;
+  private static final LogService LOGGER =
+      LogServiceFactory.getLogService(ZstdCompressor.class.getName());
+  private static final int COMPRESS_LEVEL = 3;
+
+  public ZstdCompressor() {
+  }
+
+  @Override
+  public String getName() {
+    return "zstd";
+  }
+
+  @Override
+  public byte[] compressByte(byte[] unCompInput) {
+    return Zstd.compress(unCompInput, 3);
+  }
+
+  @Override
+  public byte[] compressByte(byte[] unCompInput, int byteSize) {
+    return Zstd.compress(unCompInput, COMPRESS_LEVEL);
+  }
+
+  @Override
+  public byte[] unCompressByte(byte[] compInput) {
+    long estimatedUncompressLength = Zstd.decompressedSize(compInput);
+    return Zstd.decompress(compInput, (int) estimatedUncompressLength);
+  }
+
+  @Override
+  public byte[] unCompressByte(byte[] compInput, int offset, int length) {
+    // todo: how to avoid memory copy
+    byte[] dstBytes = new byte[length];
+    System.arraycopy(compInput, offset, dstBytes, 0, length);
+    return unCompressByte(dstBytes);
+  }
+
+  @Override
+  public byte[] compressShort(short[] unCompInput) {
+    // short use 2 bytes
+    byte[] unCompArray = new byte[unCompInput.length * 2];
+    ByteBuffer unCompBuffer = ByteBuffer.wrap(unCompArray);
+    for (short input : unCompInput) {
+      unCompBuffer.putShort(input);
+    }
+    return Zstd.compress(unCompBuffer.array(), COMPRESS_LEVEL);
+  }
+
+  @Override
+  public short[] unCompressShort(byte[] compInput, int offset, int length) {
+    byte[] unCompArray = unCompressByte(compInput, offset, length);
+    ByteBuffer unCompBuffer = ByteBuffer.wrap(unCompArray);
+    short[] shorts = new short[unCompArray.length / 2];
+    for (int i = 0; i < shorts.length; i++) {
+      shorts[i] = unCompBuffer.getShort();
+    }
+    return shorts;
+  }
+
+  @Override
+  public byte[] compressInt(int[] unCompInput) {
+    // int use 4 bytes
+    byte[] unCompArray = new byte[unCompInput.length * 4];
+    ByteBuffer unCompBuffer = ByteBuffer.wrap(unCompArray);
+    for (int input : unCompInput) {
+      unCompBuffer.putInt(input);
+    }
+    return Zstd.compress(unCompBuffer.array(), COMPRESS_LEVEL);
+  }
+
+  @Override
+  public int[] unCompressInt(byte[] compInput, int offset, int length) {
+    byte[] unCompArray = unCompressByte(compInput, offset, length);
+    ByteBuffer unCompBuffer = ByteBuffer.wrap(unCompArray);
+    int[] ints = new int[unCompArray.length / 4];
+    for (int i = 0; i < ints.length; i++) {
+      ints[i] = unCompBuffer.getInt();
+    }
+    return ints;
+  }
```

Comment:
can try following code style to convert unCompress byte result to target datatype:
(take Int for example)
```
    byte[] unCompArray = unCompressByte(compInput, offset, length);
    IntBuffer buf = ByteBuffer.wrap(unCompArray).asIntBuffer();
    int[] dest = new int[buf.remaining()];
    buf.get(dest);
    return dest;
```